Frameworks/Foundation/NSCFCalendar.mm

```diff
@@ -594,6 +594,42 @@ - (void)enumerateDatesStartingAfterDate:(NSDate*)start
     THROW_NS_IF_NULL(E_INVALIDARG, comps);
     THROW_NS_IF_FALSE(E_INVALIDARG, (opts != 0));
 
+    BOOL optionsAreInContrast = NO;
+
+    BOOL isOptionsValid = ((opts & NSCalendarMatchPreviousTimePreservingSmallerUnits) != 0);
+
+    if ((opts & NSCalendarMatchNextTimePreservingSmallerUnits) != 0) {
+        if (isOptionsValid == NO) {
+            isOptionsValid = YES;
+        } else {
+            optionsAreInContrast = YES;
+        }
+    }
+    if ((opts & NSCalendarMatchNextTime) != 0) {
+        if (isOptionsValid == NO) {
+            isOptionsValid = YES;
+        } else {
+            optionsAreInContrast = YES;
+        }
+    }
+
+    // This is a valid solo option as well as when used in conjunction with the others.
+    if (!isOptionsValid) {
+        isOptionsValid = ((opts & NSCalendarMatchStrictly) != 0);
+    }
+
+    THROW_NS_IF_FALSE_MSG(E_INVALIDARG,
+                          (isOptionsValid && !optionsAreInContrast),
+                          "Options must specify exactly one of these matching options: NSCalendarMatchPreviousTimePreservingSmallerUnits, "
+                          "NSCalendarMatchNextTimePreservingSmallerUnits, NSCalendarMatchNextTime");
```

Comment:
or not.. you do want one of these flags present, i see it now :)

---
In reply to: [73946729](https://github.com/Microsoft/WinObjC/pull/844#discussion_r73946729) [](ancestors = 73946729)